presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -1015,6 +1022,9 @@ else if (sparkException.getMessage().contains("Executor heartbeat timed out") ||
                                 sparkException.getMessage().contains("Unable to talk to the executor")) {
                             wrappedPrestoException = new PrestoException(SPARK_EXECUTOR_LOST, executionException);
                         }
```

Comment:
Would it be better to move the existing error classification logic to a separate `ErrorClassifier` to avoid this situation where some error classification happens in place here and other classification happens via the `ErrorClassifier` interface.

Reply:
Responded in the below comment.